_tools/types/main.go

```diff
@@ -0,0 +1,340 @@
+	}
+	fmt.Fprintf(os.Stderr, "%d uast:* types found in SDK\n", len(out))
+	return out
+}
+
+// analyzeFixtures goes though all fixtures, assuming the driver is cloned.
+// It updates given driverStats with results.
+func analyzeFixtures(driver *driverStats) error {
+	fixDir := path.Join(reposRootPath, driver.path, fixtureDir)
+	fixtureFiles, err := lsDir(fixDir)
+	if os.IsNotExist(err) {
+		driver.skip = true
+		return nil
+	} else if err != nil {
+		return err
+	}
+	driver.fixturesNum += len(fixtureFiles)
+
+	for _, file := range fixtureFiles {
+		if !strings.HasSuffix(file.Name(), fixtureExt) {
+			continue
+		}
+		
+		fFile, err := os.Open(path.Join(fixDir, file.Name()))
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer fFile.Close()
+
+		scanner := bufio.NewScanner(fFile)
+		for scanner.Scan() {
+			line := scanner.Text()
+			if !strings.Contains(line, "uast:") {
+				continue
+			}
+			for _, uastType := range uastNodeRe.FindAllString(line, -1) {
+				driver.fixturesUast[strings.Replace(uastType, ":", ".", 1)]++
+			}
+		}
+
+		if err := scanner.Err(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func lsDir(dir string) ([]os.FileInfo, error) {
+	fmt.Fprintf(os.Stderr, "reading %s/*%s files\n", dir, fixtureExt)
+	f, err := os.Open(dir)
+	if err != nil {
+		return nil, err
+	}
+	files, err := f.Readdir(-1)
+	f.Close()
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+// analyzeCode checks if any of the types are used by
+// this driver's package, though analyzing it's AST.
+func analyzeCode(driver *driverStats, uasts []uastType) {
+	// TODO:
+	// load package
+	// for _, typee := range uasts {
+	//   if typee.isUsedIn(package) {
+	//     driver.codeUast[typee]++
+	//   }
+	// }
+	driver.codeUast["Identifier"]++
+}
+
+func formatMarkdownTable(drivers []*driverStats, uastTypes []uastType) {
+	fmt.Print(header)
+	defer fmt.Print(footer)
+
+	allDrivers := len(drivers)
+	drs := drivers[:0] // filter drivers \wo fixtures
+	for _, x := range drivers {
+		if !x.skip {
+			drs = append(drs, x)
+		}
+	}
+	fmt.Fprintf(os.Stderr, "only %d drivers has fixtures, out of %d\n", len(drs), allDrivers)
+
+	formatMarkdownTableHeader(drs)
+	for _, typee := range uastTypes {
+		// %25s produces nice ASCII result
+		fmt.Printf("|[%s](%s)|", typee.name, goDocURL+typee.name[strings.IndexRune(typee.name, '.')+1:])
+		for _, dr := range drs {
+			fmt.Printf(" %d/%d |", dr.fixturesUast[typee.name], dr.codeUast[typee.name])
+		}
+		fmt.Println()
+	}
+}
+
+func formatMarkdownTableHeader(drivers []*driverStats) {
+	fmt.Printf("|%25s|", "")
+	for _, dr := range drivers {
+		// %5s produces nice ASCII result
+		fmt.Printf("[%s](%s)|", dr.language, dr.url)
+	}
+	fmt.Print("\n| :---------------------- |")
+	for range drivers {
+		fmt.Printf(" :-- |")
+	}
+	fmt.Println()
+}
+
+const header = `<!-- Code generated by 'make types' DO NOT EDIT. -->
+# UAST Types
+
+For every UAST type in every driver, next 2 numbers are reported:
```

Comment:
```suggestion
For every UAST type in every driver, the following two values are reported:
```

(Rationale: Many prose style manuals recommend spelling out "small" numbers longhand, and switching to digits for larger ones. Chicago, for example, recommends spelling out 0–9 or 0–100 depending on context. This is not crucial, but that's why I made that part of the recommendation).